### ISSUE TITLE
Run

### ISSUE THREAD

**Issue by Auth0rM0rgan** [Tandon-A/CycleGAN_ssim#1]
Hey @Tandon-A,

Thanks for sharing your code. I am wondering how can I run your code because in the repo you written for executing the code I need to run train_cycleGAN.py but there is not any such python file in the repo.

Also, when I tried to run train_cycleGAN_loss.py file I got this error:

>     imgA = get_image_new(trainA[countA],shape[0],shape[1])
>     IndexError:list index out of range

It would be great if you guide me to execute your code.

Thanks

**Comment by Tandon-A** [comment]
Hey @arminXerror, 

Thanks for pointing that out. I have changed the files and have updated the readme. I will update the readme in some time to tell in detail about the loss functions available for training. 

The dataset folder should have two folders named, 'trainA' containing domainA training images and 'trainB' containing domainB training images. Rest executing the train_cycleGAN_loss.py file with proper options as shown in the readme would start training a cycleGAN model.

Let me know if this works for you.

**Comment by Auth0rM0rgan** [comment]
Hey @Tandon-A, 

Thanks for updating your code and now the code is working properly. Just a few question mark in my mind

How long is going to take to train your model from scratch? because I tried other cycleGAN implementation and it takes 24 hours for 200 epochs but your code takes 30mins to train for 201 epochs.

In the paper mentioned, 6 residual layers for image size 128 and for greater than they used 9 residual layers. are you resizing all images to 128? because your implementation contains 6 residual layers.

Also, I am trying to add Tensorboard functionality to your code which is nice to see the loss function curve and generated images 

And I think your model save is not correct because it just saves one file which name is checkpoint and there are not any meta files.

Best,

**Comment by Tandon-A** [comment]
Hey @arminXerror 

I don't have a GPU and so use Google Colab for training my models. Since it is a shared resource, it takes about three to four hours to train for 200 epochs. 
You can increase the max_img (default value 100) to train for more images per epoch. That would allow the model to learn more but would demand more computation. I am not sure how this compares to other cycleGAN models which you have tested. 

I am using a model with 6 residual layers and am resizing images to 128, 128 (height, width) while loading the dataset. You can add three more residual layers and resize images to 256, 256 or larger size. 

Adding tensorboard functionality would be great. I tried that, but it made the whole training process on Google Colab very very slow. 

The model save part is working properly on Google Colab. Using the trained model, I have generated the images shown in the readme. This problem is something new which I have come across. I have checked stack overflow but I didn't find anything yet. Please let me know if you find some fix for this.
